### PR TITLE
DEV2-3634: add context to the selected-code and the diagnostics

### PR DESCRIPTION
**Pull request by yairco1990**

### CODE REVIEW

File: Common/src/main/java/com/tabnineCommon/chat/commandHandlers/context/DiagnosticsContext.kt [hunk 1]
```diff
@@ -12,8 +12,10 @@ import com.tabnineCommon.chat.commandHandlers.utils.AsyncAction
 import java.awt.Point
 import java.util.concurrent.CompletableFuture
 
+data class DiagnosticItem(val errorMessage: String, val lineNumber: Int, val lineCode: String)
+
 data class DiagnosticsContext(
-    private val diagnosticsText: String? = null,
+    private val diagnostics: List<DiagnosticItem>? = null,
```
Comment (yonip23) [inline Common/src/main/java/com/tabnineCommon/chat/commandHandlers/context/DiagnosticsContext.kt:18]: how are you handling backwards compatibility in the app and FE?

File: Common/src/main/java/com/tabnineCommon/chat/commandHandlers/context/DiagnosticsContext.kt [hunk 2]
```diff
@@ -59,5 +66,14 @@ data class DiagnosticsContext(
                 null
             }
         }
+
+        private fun getSelectedRange(editor: Editor): TextRange? {
```
Comment (yonip23) [inline Common/src/main/java/com/tabnineCommon/chat/commandHandlers/context/DiagnosticsContext.kt:70]: just make sure it actually works, im having PTSD from JB's read/write permissions...

File: Common/src/main/java/com/tabnineCommon/chat/commandHandlers/context/DiagnosticsContext.kt (hunk 1 above)
Comment (yairco1990) [reply Common/src/main/java/com/tabnineCommon/chat/commandHandlers/context/DiagnosticsContext.kt:18]: this way
https://github.com/codota/tabnine-chat-frontend/blob/main/src/services/userContextPrompts.ts#L101

**Review by yonip23 — APPROVED** [review]
🍸

**Review by dimacodota — APPROVED** [review]
lgtm

**Pull request merged**